Fails to compile with -DENABLE_WEB_SERVICE=OFF

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

yuzu git

### Description of Issue

Fails to build with `-DENABLE_WEB_SERVICE=OFF`

Very likely because of https://github.com/yuzu-emu/yuzu/pull/9238

### Expected Behavior

yuzu compiles

### Reproduction Steps

Attempt to compile yuzu with  `-DENABLE_WEB_SERVICE=OFF`

### Log File

```
qt_amiibo_settings.cpp:(.text+0x4596): undefined reference to `WebService::Client::Client(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text+0x466f): undefined reference to `WebService::Client::Client(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text+0x4721): undefined reference to `WebService::Client::GetJson(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text+0x4834): undefined reference to `WebService::Client::~Client()'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text+0x483f): undefined reference to `WebService::Client::~Client()'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text+0x51ca): undefined reference to `WebService::Client::GetImage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/bin/ld: src/yuzu/CMakeFiles/yuzu.dir/applets/qt_amiibo_settings.cpp.o: in function `QtAmiiboSettingsDialog::LoadAmiiboApiInfo(std::basic_string_view<char, std::char_traits<char> >) [clone .cold]':
qt_amiibo_settings.cpp:(.text.unlikely+0x122d): undefined reference to `WebService::Client::~Client()'
/usr/bin/ld: qt_amiibo_settings.cpp:(.text.unlikely+0x1238): undefined reference to `WebService::Client::~Client()'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### System Configuration

CPU: N/A
GPU/Driver: N/A
RAM: N/A
OS: Arch Linux x86_64

--- a/src/yuzu/applets/qt_amiibo_settings.cpp
+++ b/src/yuzu/applets/qt_amiibo_settings.cpp
@@ -13,7 +13,9 @@
 #include "input_common/drivers/virtual_amiibo.h"
 #include "input_common/main.h"
 #include "ui_qt_amiibo_settings.h"
+#ifdef ENABLE_WEB_SERVICE
 #include "web_service/web_backend.h"
+#endif
 #include "yuzu/applets/qt_amiibo_settings.h"
 #include "yuzu/main.h"
 
@@ -90,6 +92,7 @@ void QtAmiiboSettingsDialog::LoadAmiiboInfo() {
 }
 
 void QtAmiiboSettingsDialog::LoadAmiiboApiInfo(std::string_view amiibo_id) {
+#ifdef ENABLE_WEB_SERVICE
     // TODO: Host this data on our website
     WebService::Client client{"https://amiiboapi.com", {}, {}};
     WebService::Client image_client{"https://raw.githubusercontent.com", {}, {}};
@@ -134,6 +137,7 @@ void QtAmiiboSettingsDialog::LoadAmiiboApiInfo(std::string_view amiibo_id) {
     pixmap = pixmap.scaled(250, 350, Qt::AspectRatioMode::KeepAspectRatio,
                            Qt::TransformationMode::SmoothTransformation);
     ui->amiiboImageLabel->setPixmap(pixmap);
+#endif
 }
 
 void QtAmiiboSettingsDialog::LoadAmiiboData() {

